app-media-image-capture.d.ts

```diff
@@ -0,0 +1,158 @@
+/**
+ * DO NOT EDIT
+ *
+ * This file was automatically generated by
+ *   https://github.com/Polymer/gen-typescript-declarations
+ *
+ * To modify these typings, edit the source file(s):
+ *   app-media-image-capture.html
+ */
+
+/// <reference path="../polymer/types/polymer.d.ts" />
+
+/**
+ * `app-media-image-capture` implements a helpful wrapper element for the emerging
+ * [Image Capture API](https://www.w3.org/TR/image-capture/). This element enables
+ * straight-forward photographic image control and capture that cooperates nicely
+ * with other app-media elements.
+ *
+ * NOTE: Image Capture API is still under development. As of today (April 7th, 2017),
+ * the API requires a polyfill or an
+ * [Origin Trial](https://github.com/jpchase/OriginTrials/blob/gh-pages/developer-guide.md).
+ * This element will politely neglect to register itself in browsers that do not feature
+ * the appropriate ImageRecorder global constructor.
+ *
+ * Chrome and Firefox have partial support for the API behind flags. See the
+ * [ImageCapture API browser support](https://github.com/w3c/mediacapture-image/blob/gh-pages/implementation-status.md)
+ * page for details.
+ *
+ * If you wish to support browsers that do not implement the Image Capture API,
+ * please consider including a polyfill such as
+ * [this one](https://github.com/GoogleChrome/imagecapture-polyfill) in your app.
+ * Note that most polyfills will not enable full functionality, but they should give
+ * your app a reasonable fallback in browsers that do not natively support the API.
+ *
+ * ELEMENT DESIGN NOTE: Many of the properties of this element have a default
+ * configuration value of `null`. While this is weird, it is important to keep the
+ * behavior of multi-property observers in the element implementation normalized
+ * across Polymer 1.x and 2.x. Expect this aspect of the API to change as the
+ * element graduates from the current hybrid implementation to something that can
+ * rely on Polymer >=2.x observer semantics.
+ */
+interface AppMediaImageCaptureElement extends Polymer.Element {
+
+  /**
+   * The input media stream to capture images from. Note that the
+   * stream must have at least one video track in order to be a suitable
+   * source for image capture.
+   */
+  stream: MediaStream|null;
+
+  /**
+   * MediaStreams can have multiple video tracks. This property enables
+   * you to configure the index of the video track to use.
+   */
+  trackIndex: number|null|undefined;
+
+  /**
+   * The video track selected from the input MediaStream. This track will
+   * be the source for any images captured.
+   */
+  readonly videoTrack: MediaStreamTrack|null;
+
+  /**
+   * An ImageCapture instance associated with the selected video track.
+   */
+  readonly imageCapture: ImageCapture|null;
```

Comment:
I don't think the `ImageCapture` type is defined in the TypeScript standard libs, so this shouldn't compile. We could define it by manually writing an extra-types.d.ts and including it with `addReferences` in gen-tsd.json, or ignore it by putting `renameTypes: {"ImageCapture": "any"}` in the config file.

Reply:
I contributed the closure types for this constructor and its related settings object to the [closure externs](https://github.com/google/closure-compiler/blob/master/externs/browser/w3c_rtc.js). Would it be possible to make sure of generated types based on that somehow?

Alternatively, would it be possible to make a similar contribution to the TypeScript standard library?

Reply:
Yes I think we should upstream this into the `lib.dom.d.ts` typings

Reply:
That would be great. I wouldn't be opposed to doing the `any` rename for now since getting those into TS could take a while.

Reply:
It would be nice to see a first step as part of this change; maybe file an issue?